action.yml

```
@@ -90,14 +90,22 @@ runs:
       run: |
         if [[ "${{runner.os}}"  == "Linux" ]]; then
           sudo apt-get update
-          sudo apt-get install clang-format-${{ inputs.version }} clang-tidy-${{ inputs.version }} || true
+          # First try installing from default Ubuntu repositories before trying LLVM script
+          if ! sudo apt-get install clang-format-${{ inputs.version }} clang-tidy-${{ inputs.version }}; then
+            # This LLVM script will add the relevant LLVM PPA: https://apt.llvm.org/
+            wget https://apt.llvm.org/llvm.sh -O $GITHUB_ACTION_PATH/llvm_install.sh
+            chmod +x $GITHUB_ACTION_PATH/llvm_install.sh
+            sudo $GITHUB_ACTION_PATH/llvm_install.sh ${{ inputs.version }}
+            sudo apt-get install clang-format-${{ inputs.version }} clang-tidy-${{ inputs.version }} || true
```

I think we could nest the call to the installer script:
```suggestion
            if sudo $GITHUB_ACTION_PATH/llvm_install.sh ${{ inputs.version }}; then
              sudo apt-get install clang-format-${{ inputs.version }} clang-tidy-${{ inputs.version }}
            fi
```